target/riscv/op_helper_cheri.c

```diff
@@ -443,3 +444,27 @@ target_ulong HELPER(sc_c_cap)(CPUArchState *env, uint32_t addr_reg, uint32_t val
 {
     return sc_c_impl(env, addr_reg, val_reg, /*offset=*/0, GETPC());
 }
+
+target_ulong HELPER(cloadtags)(CPUArchState *env, uint32_t cb)
+{
+    GET_HOST_RETPC();
+    const cap_register_t *cbp = get_capreg_0_is_ddc(env, cb);
+    target_ulong cursor = cap_get_cursor(cbp);
+
+    if (!cbp->cr_tag) {
+        raise_cheri_exception(env, CapEx_TagViolation, cb);
+    } else if (!cap_is_unsealed(cbp)) {
+        raise_cheri_exception(env, CapEx_SealViolation, cb);
+    } else if (!(cbp->cr_perms & CAP_PERM_LOAD)) {
+        raise_cheri_exception(env, CapEx_PermitLoadViolation, cb);
+    } else if (!(cbp->cr_perms & CAP_PERM_LOAD_CAP)) {
+        raise_cheri_exception(env, CapEx_PermitLoadCapViolation, cb);
```

Comment:
Other loads are tag-stripping in this case. I know it's not useful for cloadtags, but it would match Morello.
Maybe worth adding a comment such as "CLoadTags raises an exception instead of stripping tags"?